### ISSUE TITLE
Precision issue with LinearAlgebra Exercise 14 validation

### ISSUE THREAD

**Issue by ivylee** [microsoft/QuantumKatas#225]
Following is my answer to linear algebra exercise 14, it never passes validation, possibly due to a precision issue:

```
@exercise
def find_eigenvector(a : Matrix, x : float) -> Matrix:
    m, n = a[0]
    p, q = a[1]
    print("input matrix", m, n, p, q)
    
    if (m - x) == 0:
        if n == 0:
            if p == 0 and (q - x) == 0:
                return [[1], [1]]
            elif p == 0:
                return [[1], [0]]
            elif (q - x) == 0:
                return [[0], [1]]
        else:
            return [[1], [0]]
    if p == 0:
        if (q - x) == 0:
            if (m - x) == 0 and n == 0:
                return [[1], [1]]
            elif (m - x) == 0:
                return [[1], [0]]
            elif n == 0:
                return [[0], [1]]
        else:
            return [[1], [0]]
    
    res = [[n], [x - m]]
    print("=> matrix_mult", matrix_mult(a, res))
    print("=> scalar_mult", scalar_mult(x, res))
    
    return res
```

Output

```
input matrix 2 0 0 0
input matrix 0 0 0 2
input matrix 3 -2 -3 2
=> matrix_mult [[0], [0]]
=> scalar_mult [[0], [0]]
input matrix (1.410044130618676+3.4335219182928984j) (3.754740821442419-3.844639920338417j) (-1.3835599382959733-3.9144870184937854j) (-3.6290973002432683+3.692620479238418j)
=> matrix_mult [[(-0.8412196457268912-2.04680458238229j)], [(-1.6445865370993324-0.1502656593596896j)]]
=> scalar_mult [[(-0.841219645726889-2.046804582382289j)], [(-1.6445865370993324-0.15026565935968822j)]]
Wrong eigenvector!
Eigenvalue: 0.163-0.378j

   |  1.410+3.434i  3.755-3.845i |
A: | -1.384-3.914i -3.629+3.693i |

              |  3.755-3.845i |
You returned: | -1.247-3.812i |

Try again!
```

`matrix_mult` output and `scaler_mult` outputs are very close, but the testing module would rule it wrong (https://github.com/microsoft/QuantumKatas/blob/master/tutorials/LinearAlgebra/testing.py#L157). The precision issue seems to only occur with complex value inputs.

**Comment by tcNickolas** [comment]
True, I tried a different solution and it also failed precision check. The first condition on that line is not necessary, `act[i] != approx(exp[i])` should be sufficient (currently it fails as soon as exact equality doesn't happen). 

Do you want to send a pull request to fix this? I would like to also include the matrix product and the scalar product in the error message for this task, so that the learner doesn't have to modify their code to print them like you had to in your solutions. If not, I'll get to it over the weekend or after it.

Thank you for catching this!

**Comment by tcNickolas** [comment]
The fix will get into master once I merge #237. 
Thank you again for reporting this!
